xpad.c

```diff
@@ -1971,6 +2064,26 @@ static int xpad_probe(struct usb_interface *intf, const struct usb_device_id *id
 
 	usb_set_intfdata(intf, xpad);
 
+	/* Special packet type detection */
+	if (le16_to_cpu(udev->descriptor.idVendor) == 0x045e) { /* Microsoft controllers */
+		if (le16_to_cpu(udev->descriptor.idProduct) == 0x02e3) { /* Elite 1 controller */
+			/* The original elite controller always uses the oldest type of extended packet */
+			xpad->ptype = PTYPE_EONE;
+		} else if (le16_to_cpu(udev->descriptor.idProduct) == 0x0b00) { /* Elite 2 controller */
+			/* The elite 2 controller has seen multiple packet revisions. These are tied to specific firmware versions */
+			if (le16_to_cpu(udev->descriptor.bcdDevice) < 0x0500) {
+				/* This is the format that the Elite 2 used prior to the BLE update */
+				xpad->ptype = PTYPE_ETWOOLD;
```

Comment:
I couldn't find any strong evidence of there being any earlier packet formats for **wired** connections, so we just assume that the pre-BLE behavior is the oldest one.

I know that earlier firmwares once had a bigger packet format when using _Bluetooth_, but it's not clear if that was ever something that also affected the wired connection. It seems that people just weren't documenting that at the time...

I'd _love_ it if we could verify this assumption by finding someone who owns early a XBE2 controller on the original firmware, but I doubt that's going to happen. Odds are... if this ever is an issue, we can patch it with the help of whoever reports the bug.